src/main/java/picard/nio/PicardBucketUtils.java

```diff
@@ -67,19 +65,48 @@ public static PicardHtsPath getTempFilePath(final String directory, String prefi
         }
     }
 
-    // For local temp file, directory should be null.
+    /**
+     * This overload of getTempFilePath takes the directory of type PicardHtsPath instead of String.
+     */
+    public static PicardHtsPath getTempFilePath(final PicardHtsPath directory, String prefix, final String extension){
+        return getTempFilePath(directory.getURIString(), prefix, extension);
+    }
+
+    /**
+     * Calls getTempFilePath without the prefix.
+     */
     public static PicardHtsPath getTempFilePath(String directory, String extension){
         return getTempFilePath(directory, "", extension);
     }
 
+    public static PicardHtsPath getLocalTempFilePath(final String prefix, final String extension){
+        return getTempFilePath((String) null, prefix, extension);
+    }
+
+    /**
+     * Creates a path to a "directory" on a Google Cloud System filesystem with a randomly generated URI.
+     * Since the notion of directories does not exist in GCS, it creates a path to a URI ending in "/".
+     * Calling this method will not create a directory/file on GCS. It merely returns a path to a non-directory.
```

Comment:
This line is super-confusing:

> It merely returns a path to a non-directory.

since this method is supposed to be creating a directory. I would suggest maybe saying "it creates an object that can be used as a parent container for other objects" or something similar instead.

Reply:
Thanks, I attempted to rewrite this.